app/src/main/java/com/duckduckgo/app/history/HistoryRepository.kt

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.history
+
+import com.duckduckgo.app.history.store.HistoryDao
+import com.duckduckgo.common.utils.CurrentTimeProvider
+import io.reactivex.Single
+
+interface HistoryRepository {
+    fun getHistoryObservable(): Single<List<HistoryEntry>>
+    fun saveToHistory(
+        url: String,
+        title: String?,
+        query: String?,
+        isSerp: Boolean,
```

Comment:
I would say isSerp can be calculated inside the implementation and hidden from the interface.

1) Seems to me we are exposing the requirement of knowing if a visited url is serp in the interface. 2) thinking all the possible clients of this api, not forcing them to calculate it's simpler for them. So a nice api.

Reply:
I don't think we should consider HistoryRepository to be the api, but rather SaveToHistory. If we look at it that way, then the API is pretty simple, we just ask clients for an URL and a title, and the implementation does all the calculations needed

Reply:
"I don't think we should consider HistoryRepository to be the api, but rather SaveToHistory." 

yeah, I saw there was another interface later on the review. All good if this is internal 👍